### PR TITLE
Update Storage to handle Event

### LINKED ISSUE #116: Add storage logic for Events
## Description

`Event` data should be persisted in `addressbook.json`.

- [x] Add `JsonAdaptedEvent`
- [x] Add `JsonAdaptedEvent` to `JsonSerializableAddressBook`
- [x] Add tests

### PR DESCRIPTION
**Pull request by casperngeen**
Closes #116 

I also added some updates to `Description` and `Event`:
- For `Description`, I updated the validation check to a `REGEX` which checks for the presence of at least one non-whitespace character
- For `Event`, I changed the type of `clients` and `vendors` from `ArrayList` to the abstract version `List`, to allow for more flexible usage

### CODE REVIEW

File: src/main/java/seedu/ddd/model/event/Description.java [hunk 1]
```diff
@@ -28,7 +31,7 @@ public Description(String description) {
      * Returns true if a given string is a valid description.
      */
     public static boolean isValidDescription(String test) {
-        return !test.equals(EMPTY_STRING);
+        return test.matches(VALIDATION_REGEX);
```
Comment (FYL2003) [inline src/main/java/seedu/ddd/model/event/Description.java:34]: This change will make the design more flexible.

File: src/main/java/seedu/ddd/model/AddressBook.java
Comment (FYL2003) [inline src/main/java/seedu/ddd/model/AddressBook.java:1]: The change will provide a unique ID for both contact and event.

**Review by FYL2003 — COMMENTED** [review]
The added and modified part LGTM.

File: src/main/java/seedu/ddd/storage/JsonAdaptedEvent.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import seedu.ddd.model.contact.vendor.Vendor;
+import seedu.ddd.model.event.Description;
+import seedu.ddd.model.event.Event;
+import seedu.ddd.model.event.EventId;
+
+class JsonAdaptedEvent {
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Event's %s field is missing!";
+    private final int eventId;
+    private final String description;
+    private final List<JsonAdaptedClient> clients = new ArrayList<>();
```
Comment (Teddayz) [inline src/main/java/seedu/ddd/storage/JsonAdaptedEvent.java:21]: I think changing ArrayList<> to List<> to allow more flexibility was good!

**Review by Teddayz — APPROVED** [review]
LGTM! well written :)

**Comment by Teddayz** [comment]
LGTM

**Pull request merged**